ext/curl/interface.c

```diff
@@ -2229,13 +2233,22 @@ static inline int build_mime_structure_from_hash(php_curl *ch, zval *zpostfields
 				cb_arg->filename = zend_string_copy(postval);
 				cb_arg->stream = NULL;
 
+				if ((stream = php_stream_open_wrapper(ZSTR_VAL(postval), "rb", STREAM_MUST_SEEK, NULL))) {
```

Comment:
Now that we open the stream here anyway, does it really make sense to keep the lazy-opening behavior for the stream in read_cb?

Reply:
At least for typical cases it doesn't, so I've changed that.